docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -0,0 +1,286 @@
+[[size-your-shards]]
+== How to size your shards
+++++
+<titleabbrev>Size your shards</titleabbrev>
+++++
+
+To protect against hardware failure and increase capacity, {es} stores copies of
+an index’s data across multiple shards on multiple nodes. The number and size of
+these shards can have a significant impact on your cluster's health. One common
+problem is _oversharding_, a situation in which a cluster with a large number of
+shards has degraded performance.
+
+[discrete]
+[[create-a-sharding-strategy]]
+=== Create a sharding strategy
+
+The best way to prevent oversharding and other shard-related performance issues
+is to create a sharding strategy. A sharding strategy helps you determine and
+maintain the optimal number of shards for your cluster while limiting the size
+of those shards.
+
+Unfortunately, there is no one-size-fits-all sharding strategy. A strategy that
+works in one environment may not scale in another. A good sharding strategy must
+account for your infrastructure, use case, and performance expectations.
+
+The best way to create a sharding strategy is to benchmark your production data
+on production hardware using the same queries and indexing loads you'd see in
+production. For our recommended methodology, watch the
+https://www.elastic.co/elasticon/conf/2016/sf/quantitative-cluster-sizing[quantitative
+cluster sizing video]. As you test different shard configurations, use {kib}'s
+{kibana-ref}/elasticsearch-metrics.html[{es} monitoring] to track your cluster's
+performance.
+
+The following sections provide some reminders and guidelines you should consider
+when designing your sharding strategy. If your cluster has shard-related
+problems, see <<fix-an-oversharded-cluster>>.
+
+[discrete]
+[[shard-size-reminders]]
+=== Things to remember
+
+Keep the following considerations in mind when building your sharding strategy.
+
+[discrete]
+[[single-thread-per-shard]]
+==== Searches run on a single thread per shard
+
+Most searches hit multiple shards. Each shard runs the search on a single
+thread. While a shard can run multiple concurrent searches, searches across a
+large number of shards can deplete a node's <<modules-threadpool,search
+threadpool>>. This can result in low throughput and slow search speeds.
+
+[discrete]
+[[each-shard-has-overhead]]
+==== Each shard has overhead
+
+Every shard uses file handles, memory, and CPU resources. In most cases, a small
+set of large shards uses fewer resources than a large set of small shards.
+
+Segments play a large role in a shard's resource usage. Most shards contain
+several segments, which store its index data. For faster searches, segment
+metadata is kept in <<heap-size,heap memory>>. As a shard grows, its segments
+are <<index-modules-merge,merged>> into fewer, larger segments. As the number of
+segments decreases, less metadata needs to be kept in heap memory.
```

Comment:
Note for reviewers:

Is this still the case with our recent heap improvements in 7.7?
https://www.elastic.co/guide/en/elasticsearch/reference/7.7/release-highlights.html#_significant_reduction_of_heap_usage_of_segments